cmd/kubeadm/app/util/apiclient/idempotency_test.go

```diff
@@ -237,3 +240,227 @@ func TestMutateConfigMapWithConflict(t *testing.T) {
 		t.Fatalf("ConfigMap mutation with conflict was invalid, has: %q", cm.Data["key"])
 	}
 }
+
+func TestGetConfigMapWithShortRetry(t *testing.T) {
+	type args struct {
+		client    clientset.Interface
+		namespace string
+		name      string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *v1.ConfigMap
+		wantErr bool
+	}{
+		{
+			name: "ConfigMap exists",
+			args: args{
+				client:    newMockClientForTest(t, "default", "foo", "ConfigMap"),
+				namespace: "default",
+				name:      "foo",
+			},
+			want: &v1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       configMapName,
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "default",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "ConfigMap does not exist",
+			args: args{
+				client:    fake.NewSimpleClientset(),
+				namespace: "default",
+				name:      "foo",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetConfigMapWithShortRetry(tt.args.client, tt.args.namespace, tt.args.name)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetConfigMapWithShortRetry() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("GetConfigMapWithShortRetry() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCreateOrUpdateClusterRole(t *testing.T) {
+	type args struct {
+		client      clientset.Interface
+		clusterRole *rbac.ClusterRole
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "ClusterRole does not exist",
+			args: args{
+				client: fake.NewSimpleClientset(),
+				clusterRole: &rbac.ClusterRole{
```

Comment:
create this object only one time near the  top of TestCreateOrUpdateClusterRole as it's the same for both test cases?